src/pilet-build.test.ts

```diff
@@ -214,4 +214,144 @@ runTests('pilet-build', ({ test, setup }) => {
       });
     },
   );
+
+  // 'can build a pilet with default target ./dist',
+  test(
+    'from-sample-piral-with-default-target',
+    'can build a pilet with default target ./dist/index.js',
+    ['pilet.v2', 'build.pilet'],
+    async (ctx) => {
+      await ctx.run(`npx pilet build --target`);
+
+      await ctx.assertFiles({
+        'dist/index.js'(content: string) {
+          expect(content).not.toBe('');
+          expect(content).toContain('//@pilet v:2');
+          expect(content).toContain('System.register(');
+          expect(content).not.toContain('currentScript.app=');
+          expect(content).not.toContain('require("react")');
+        },
+      });
+    },
+  );
+
+  // 'from-sample-piral-with-different-target' --done,
+  test(
+    'from-sample-piral-with-different-target',
+    'can build a pilet with a different target',
+    ['pilet.v2', 'build.pilet'],
+    async (ctx) => {
+      await ctx.run(`npx pilet build --target './different-target/index.js'`);
+
+      await ctx.assertFiles({
+        dist: false,
+        'different-target/index.js'(content: string) {
+          expect(content).not.toBe('');
+          expect(content).toContain('//@pilet v:2');
+          expect(content).toContain('System.register(');
+          expect(content).not.toContain('currentScript.app=');
+          expect(content).not.toContain('require("react")');
+        },
+        'different-target/main.css': true,
+        'dist/index.js': false,
+      });
+    },
+  );
+
+  // 'can build a pilet with minify --done'
+  test('from-sample-piral-with-minify', 'can build a pilet with minify', ['pilet.v2', 'build.pilet'], async (ctx) => {
+    await ctx.run(`npx pilet build --minify`);
+
+    await ctx.assertFiles({
+      'dist/index.js'(content: string) {
+        expect(content).not.toBe('');
+        expect(content).not.toContain('var moduleUrl');
+      },
+    });
+  });
+
+  // 'can build a pilet without minify' --done,
+  test(
+    'from-sample-piral-without-minify',
+    'can build a pilet without minify',
+    ['pilet.v2', 'build.pilet'],
+    async (ctx) => {
+      await ctx.run(`npx pilet build --no-minify`);
+
+      await ctx.assertFiles({
+        'dist/index.js'(content: string) {
+          expect(content).not.toBe('');
+          expect(content).toContain('var moduleUrl');
```

Comment:
That looks to be rather bundler specific.

My suggestion here is:

1. Before running that test modify the src/index.tsx. Add some content here, something like an internal function `function foobar1234() { return 42; }` which is then used inside the `setup` function (e.g., `console.log(foobar())`).
2. In the minified version there shouldn't be any `foobar1234`
3. In the non-minified version we should see `foobar1234`